### DISCUSSION TITLE
Can't use Copilot with Proxy

### DISCUSSION THREAD

**Discussion by GiorgosAlexakis** [community/community#13113] [Copilot Conversations] [answered]
Traffic at my work has to pass through a **proxy**.

Visual Studio reads the proxy configuration from the following variables: 

```
$http_proxy=http://[proxy-server-ip-or-dns-name]:[Port-Number] $https_proxy=$http_proxy
``` 

Even though, I can use other extensions with the proxy (ex. I can commit to a Github repo from Visual Studio), **the copilot extension throws this error:**  `GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 140.82.121.5:443"`

**Comment by davecheney** [comment]
I'm sorry, Copilot does not support proxies at the moment. You may find some information in this thread, https://github.com/github/feedback/discussions/8866

**Reply by GiorgosAlexakis** [reply]
Okay I understand thanks for the quick response, it would be a nice feature to have . I will look into the thread too.

**Reply by Matti-Koopa** [reply]
Is there an active issue for this feature request? I would like to get updated.

**Reply by jnvshubham7** [reply]
**how to use this solutoin please give me brief about that?**
```
copilot-proxy.githubusercontent.com
api.github.com (Also used for Self-hosted runners)
```


This error have shown in vs code when we trying to connect copilot on my collage internet which will use proxy system
`[ERROR] [default] [2022-06-16T05:43:59.030Z] GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"`

how to solve this problem ?
please reply me

**Reply by ooyamatakehisa** [reply]
The same error has occured for me. I'm also using copilot with my university network that uses proxy.

**Reply by mabusharikh** [reply]
This is insane.. you'll find that most of your clients are corporates who mandate usage of proxies! I have tried the plug in in Rider, Intellij, Pycharm and VsCode... and it failed to work in ALL of them.. it doesn't even respect the env. variables for proxy HTTP_PROXY etc. This is not doing your product any good.. can you please provide an ETA for getting this resolved?

**Reply by austynrowat** [reply]
Was this just fixed as of Aug 8? Mine randomly started working.

**Reply by systematicguy** [reply]
There is a version GitHub Copilot 1.1.29.1869 which has arrived on Aug 09, 2022 with the promise of "Basic support for using a proxy server" but I am not going to the bleeding edge now, will patiently wait for others here to test it. :) 
https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/206547

**Reply by quinten-goens** [reply]
Nice to hear! I might test it this week

**Reply by austynrowat** [reply]
So far it's been working great on proxy for me! 👍

**Comment by abir-taheer** [comment]
Would like to provide an update from personal experience. I never got it working with my organization's proxy in VSCode but there is also an equivalent extension for IntelliJ Idea and that does in fact work with the proxy configuration

**Reply by ZhuangJiayu** [reply]
VSCode is still not working with my org's proxy. How did you solve that, or it just works?

**Comment by jnvshubham7** [comment]
This error have shown in vs code when we trying to connect copilot on my collage internet which will use proxy system  `[ERROR] [default] [2022-06-16T05:43:59.030Z] GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"`
  how to solve this problem ? please reply me

**Comment by gittar** [comment]
could someone from @github please make a statement wether it is planned to support proxies? Thank you!

**Comment by austynrowat** [comment]
With the help of the proxy team at my company I was able to get a workaround for this issue.

Using a tool proxycap (http://www.proxycap.com/download.html) I added my corporate proxy and then added both copilot-proxy.githubusercontent.com and api.github.com as rules.

This effectively routes the github copilot data through your corporate proxy.

**Reply by angel-star** [reply]
听我说👂👂👂谢谢你🙏🙏🙏因为有你👉👉👉温暖了四季🌈🌈🌈

**Reply by eatingfirst-t** [reply]
Have you solved it?can you help me？

**Reply by angel-star** [reply]
> Have you solved it?can you help me？

https://github.com/github-community/community/discussions/16960#discussioncomment-2902904

**Reply by angel-star** [reply]
> Have you solved it?can you help me？

i made it by proxifier not proxycap ,not quite but almost.

**Comment by gittar** [comment]
I couldn't agree more. The adoption of github copilot  could be massively increased by providing a solution for corporate proxies.--Sent from my Android phone with WEB.DE Mail. Please excuse my brevity.On 14/07/2022, 15:17 mabusharikh ***@***.***> wrote:

   
  This is insane.. you'll find that most of your clients are corporates who mandate usage of proxies! I have tried the plug in in Rider, Intellij, Pycharm and VsCode... and it failed to work in ALL of them.. it doesn't even respect the env. variables for proxy HTTP_PROXY etc. This is not doing your product any good.. can you please provide an ETA for getting this resolved? 
  —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
 
  
  
 
 I couldn't agree more. The adoption of github copilot  could be massively increased by providing a solution for corporate proxies.--Sent from my Android phone with WEB.DE Mail. Please excuse my brevity.On 14/07/2022, 15:17 mabusharikh ***@***.***> wrote:

   
  This is insane.. you'll find that most of your clients are corporates who mandate usage of proxies! I have tried the plug in in Rider, Intellij, Pycharm and VsCode... and it failed to work in ALL of them.. it doesn't even respect the env. variables for proxy HTTP_PROXY etc. This is not doing your product any good.. can you please provide an ETA for getting this resolved?

**Comment by houming818** [comment]
+1

**Comment by sbleden** [comment]
+1

**Comment by dartyuhov** [comment]
+1

**Comment by davecheney** [comment, marked as answer]
Please see https://github.com/community/community/discussions/29127
